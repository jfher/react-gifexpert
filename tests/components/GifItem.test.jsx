const { render, screen } = require("@testing-library/react")
const { GifItem } = require("../../src/components")

describe('Testsuit for GifItem Component', () => { 
    const title = 'One Punch Man';
    const url = 'http://thisisafakeurl.com/';
    test('should match the snapshot', () => {
        const { container } =  render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('should show errors when proptypes are missing', () => {
        render(<GifItem />);
        expect(screen.getByRole('paragraph').innerHTML).not.toBeTruthy();
        expect(screen.getByRole('img').querySelector('source')).not.toBeTruthy();
    })
    
    test('should show the image with the url and alt attribs defined', () => {
        render(<GifItem title={title} url={url}/>);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show the title passed via props inside a p tag', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByRole('paragraph').innerHTML).toBe(title);
    })
    
})