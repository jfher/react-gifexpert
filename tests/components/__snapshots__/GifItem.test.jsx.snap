// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Testsuit for GifItem Component should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man"
      src="http://thisisafakeurl.com/"
    />
    <p>
      One Punch Man
    </p>
  </div>
</div>
`;
